SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/UnresolvedReferenceTest/UnresolvedReferenceTest.cpp

```diff
@@ -0,0 +1,76 @@
+
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "UnresolvedReferenceTest.h"
+#include "UnresolvedReferenceTestActor.h"
+#include <Runtime/Core/Public/Algo/Count.h>
+
+AUnresolvedReferenceTest::AUnresolvedReferenceTest()
+	: Super()
+{
+	Author = TEXT("Ajanth");
+	Description = TEXT(
+		"Tests what happens when structs with references to actors whose entity have not been created yet are replicated. \
+						The unresolved references should evenetually be resolved");
+}
+
+void AUnresolvedReferenceTest::PrepareTest()
+{
+	Super::PrepareTest();
+
+	// Create actor holding references to the other actors
+	AddStep(TEXT("MakeReferenceActor"), FWorkerDefinition::Server(1), nullptr, [this]() {
+		AUnresolvedReferenceTestActor* RefsActor =
+			GetWorld()->SpawnActor<AUnresolvedReferenceTestActor>(FVector::ZeroVector, FRotator::ZeroRotator);
+		RegisterAutoDestroyActor(RefsActor);
+
+		FinishStep();
+	});
+
+	// Wait for actor which will hold references to replicate to the clients
+	AddStep(
+		TEXT("ClientWaitForReplication"), FWorkerDefinition::AllClients, nullptr,
```

Comment:
A nicer way of doing this would be to have a require that passes when the number of AUnresolvedReferenceTestActor is 1. This assumes the actor will be replicated in 0.5 seconds which it probably will but...

Reply:
+1